### DISCUSSION TITLE
Help wanted:How to use the options parameter in the in the openDatabase() method.

### DISCUSSION THREAD

**Discussion by Haadi-A** [tekartik/sqflite#892] [Q&A] [answered]
Hello, thanks very much for this excellent package if any of the creators are reading this.  I'm very new to programming and dart as a whole and I'm trying to open a database using the databaseFactoryFfiObj.openDatabase() method and I'm getting some errors when I pass an onCreate argument saying that there is no parameter named onCreate. When I checked, what I was getting was that I was supposed to use the options parameter rather, so I tried to use that but I didn't know how to use the onCreate in that parameter. If anybody could help me resolve this I would be very grateful, thanks in advance.

**Comment by alextekartik** [comment]
sqflite assumes a decent knowledge of flutter and SQLite in general and I know the doc is not very good. Maybe starting with SQLite is not the easiest (shared preferences could be easier so start with if you just need to save a few data).

There are some example of opening/versioning here: https://github.com/tekartik/sqflite/blob/master/sqflite/doc/opening_db.md#migration

The API is slightly different when you use the factory (basically all the parameters are in the options).

> I'm getting some errors when I pass an onCreate argument saying that there is no parameter named onCreate
> If anybody could help me resolve 

We could help you better if you show what you are doing (some code) and the exact error (from the console) you are getting as we cannot guess what you are doing wrong nor what error you are seeing.

One important note: if you use onCreate, you have to specify a version too.

**Reply by Haadi-A** [reply]
Thanks for replying @alextekartik sorry for not including more info I will do it below, you see when I use the openDatabase() which is imported from the sqflite  for mobile it allows me to pass the oncreate function, the error only pops when I use the openDatabase method from sqflite_common_ffi. Here is the image of the code giving the error:

  
![errors](https://user-images.githubusercontent.com/77079200/199030697-744950f3-9da6-4e0f-a68c-cf7ad5a83684.png)

Here is the error message I received:

![errror msg](https://user-images.githubusercontent.com/77079200/199030801-78cdf19c-ffe9-4967-8e1e-ad6c030bdc84.png)

Agin thanks very much for replying.

**Comment by alextekartik** [comment, marked as answer]
Ok I see, indeed the [`DatabaseFactory.openDatabase` API](https://pub.dev/documentation/sqflite_common/latest/sqlite_api/DatabaseFactory/openDatabase.html) is slightly different then the sqflite [openDatabase API](https://pub.dev/documentation/sqflite/latest/sqflite/openDatabase.html) that you might have seen in some example. All the parameters must be set in an options object. You should do something like:
```dart
var db = await dbFactoryFfiObj.openDatabase(path,
          options: OpenDatabaseOptions(
              version: 1,
              onCreate: (db, version) async {
                // ... your onCreate implementation here
              }));
```

**Reply by Haadi-A** [reply]
Thanks very much @alextekartik it worked.
